File: java/chaincue-real-estate/src/main/java/com/chaincuerealestate/chaincuerealestate/domains/Broker.java
package com.chaincuerealestate.chaincuerealestate.domains;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity(name = "broker")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Broker {
    @Id
    private String id;
    private String name;
    private String phoneNumber;
    private String email;

    public static Broker create(String email) {
        return new Broker(
                UUID.randomUUID().toString(),
                "",
                "",
                email
        );
    }
}

File: kotlin/chaincue-real-estate/src/main/kotlin/com/example/chaincuerealestate/domains/Broker.kt
package com.example.chaincuerealestate.domains

import jakarta.persistence.Entity
import jakarta.persistence.Id
import java.util.UUID

@Entity(name = "broker")
data class Broker(
    @Id
    var id: String,
    var name: String,
    var phoneNumber: String,
    var email: String,
) {
    companion object {
        fun create(email: String): Broker {
            return Broker(
                id = UUID.randomUUID().toString(),
                name = "",
                phoneNumber = "",
                email = email
            )
        }
    }
}

